### PR TITLE
[IMM32] Implement ImmLoadLayout

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing Japanese input...
JIRA issue: [CORE-11700](https://jira.reactos.org/browse/CORE-11700)

## Proposed changes

- Implement `imm32!ImmLoadLayout` function.

### CODE REVIEW

File: dll/win32/imm32/imm.c [hunk 1]
```diff
@@ -105,6 +111,57 @@ Imm32NotifyAction(HIMC hIMC, HWND hwnd, DWORD dwAction, DWORD_PTR dwIndex, DWORD
     return TRUE;
 }
 
+HKL WINAPI ImmLoadLayout(DWORD dwLayout, PIMEINFOEX pImeInfoEx)
+{
+    DWORD cbData;
+    UNICODE_STRING UnicodeString;
+    HKEY hLayoutKey = NULL, hLayoutsKey = NULL;
+    LONG error;
+    HRESULT hr;
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:120]:
```suggestion
    NTSTTUS Status;
```

File: dll/win32/imm32/imm.c [hunk 2]
```diff
@@ -105,6 +111,57 @@ Imm32NotifyAction(HIMC hIMC, HWND hwnd, DWORD dwAction, DWORD_PTR dwIndex, DWORD
...
+    TRACE("ImmLoadLayout(0x%08lX, %p)\n", dwLayout, pImeInfoEx);
+
+    if (IS_IME_HKL(dwLayout) ||
+        !g_psi || (g_psi->dwSRVIFlags & SRVINFO_METRICS) == 0 ||
+        ((PW32CLIENTINFO)NtCurrentTeb()->Win32ClientInfo)->W32ClientInfo[0] & 2)
+    {
+        UnicodeString.Buffer = szLayout;
+        UnicodeString.MaximumLength = sizeof(szLayout);
+        hr = RtlIntegerToUnicodeString(dwLayout, 16, &UnicodeString);
+        if (FAILED(hr))
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:132]:
```suggestion
        Status = RtlIntegerToUnicodeString(dwLayout, 16, &UnicodeString);
        if (!NT_SUCCESS(Status))
```

File: dll/win32/imm32/imm.c [hunk 3]
```diff
@@ -105,6 +111,57 @@ Imm32NotifyAction(HIMC hIMC, HWND hwnd, DWORD dwAction, DWORD_PTR dwIndex, DWORD
...
+    HKEY hLayoutKey = NULL, hLayoutsKey = NULL;
+    LONG error;
+    HRESULT hr;
+    WCHAR szLayout[MAX_PATH];
+
+    TRACE("ImmLoadLayout(0x%08lX, %p)\n", dwLayout, pImeInfoEx);
+
+    if (IS_IME_HKL(dwLayout) ||
+        !g_psi || (g_psi->dwSRVIFlags & SRVINFO_METRICS) == 0 ||
+        ((PW32CLIENTINFO)NtCurrentTeb()->Win32ClientInfo)->W32ClientInfo[0] & 2)
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:127]: Magic flag?

File: dll/win32/imm32/imm.c [hunk 4]
```diff
@@ -105,6 +111,57 @@ Imm32NotifyAction(HIMC hIMC, HWND hwnd, DWORD dwAction, DWORD_PTR dwIndex, DWORD
     return TRUE;
 }
 
+HKL WINAPI ImmLoadLayout(DWORD dwLayout, PIMEINFOEX pImeInfoEx)
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:114]: Shouldn't the first parameter be a HKL too?

Comment (katahiromz) [reply dll/win32/imm32/imm.c]: Made it `HKL`. OK.

File: dll/win32/imm32/imm.c (hunk 2 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: OK.

File: dll/win32/imm32/imm.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: OK.

File: dll/win32/imm32/imm.c (hunk 3 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c:127]: This is unknown flag.

**Pull request merged**